src/collection-view.js

```diff
@@ -125,11 +125,91 @@ const CollectionView = Backbone.View.extend({
     }
   },
 
-  // get the child view by model it holds, and remove it
-  _onCollectionRemove(model) {
-    const view = this.children.findByModel(model);
-    this.removeChildView(view);
-    this._checkEmpty();
+  // Handle collection update model removals
+  _onCollectionUpdate(collection, options) {
+    const changes = options.changes;
+    this._removeChildViews(changes.removed);
+  },
+
+  // Remove the child views and destroy them.
+  // This function also updates the indices of later views
+  // in the collection in order to keep the children in sync with the collection.
+  // "modelsOrViews" can be an array of mixed models or views that
+  // will be removed and destroyed from the CollectionView
+  _removeChildViews(modelsOrViews, {checkEmpty} = {}) {
+    const shouldCheckEmpty = checkEmpty !== false;
+
+    // Used to determine where to update the remaining
+    // sibling view indices after these views are removed.
+    const removedViews = this._getRemovedViews(modelsOrViews);
+
+    if (!removedViews.length) {
+      return;
+    }
+
+    this.children._updateLength();
+
+    // decrement the index of views after this one
+    this._updateIndices(removedViews, false);
+
+    if (shouldCheckEmpty) {
+      this._checkEmpty();
+    }
+  },
+
+  // "modelsOrViews" can be an array of mixed models or views that
+  // will be removed and destroyed from the CollectionView
+  // Returns the view that will be used for re-indexing views
+  // through CollectionView#_updateIndices.
+  _getRemovedViews(modelsOrViews) {
+
+    // the "memo" has to be `undefined` to make sure
```

Comment:
this comment doesn't look right now.. it's an array.

Also should it be `removingViews` instead of `removingsView`?